format-avro/src/test/java/io/cdap/plugin/format/avro/input/AvroInputFormatProviderTest.java

```diff
@@ -70,4 +71,68 @@ public void testSchemaDetection() throws IOException {
+    org.apache.avro.Schema avroSchema = new org.apache.avro.Schema.Parser().parse(expectedSchema.toString());
+    File avroFile = new File(TMP_FOLDER.newFolder(), "testPathFieldAddition.avro");
+    GenericRecord record = new GenericRecordBuilder(avroSchema)
+      .set("id", 0)
+      .set("name", "alice")
+      .set("score", 1.0d)
+      .set("pathField", avroFile.getAbsolutePath())
+      .build();
+    DatumWriter<GenericRecord> datumWriter = new GenericDatumWriter<>(avroSchema);
+    DataFileWriter<GenericRecord> dataFileWriter = new DataFileWriter<>(datumWriter);
+    dataFileWriter.create(avroSchema, avroFile);
+    dataFileWriter.append(record);
+    dataFileWriter.close();
+
+    // call detect schema
+    AvroInputFormatProvider.Conf conf = new AvroInputFormatProvider.Conf("pathField");
+    AvroInputFormatProvider formatProvider = new AvroInputFormatProvider(conf);
+    FormatContext formatContext = new FormatContext(new MockFailureCollector(), null);
+    SchemaDetector schemaDetector = new SchemaDetector(formatProvider);
+    Schema schema = schemaDetector.detectSchema(avroFile.getPath(), formatContext, Collections.emptyMap());
+    Assert.assertEquals(expectedSchema, schema);
+  }
+
+  @Test
+  public void testPathFieldConflict() throws IOException {
+    // write avro file
+    Schema expectedSchema = Schema.recordOf("Avro",
+      Schema.Field.of("id", Schema.of(Schema.Type.INT)),
+      Schema.Field.of("name", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+      Schema.Field.of("score", Schema.of(Schema.Type.DOUBLE)),
+      Schema.Field.of("test", Schema.of(Schema.Type.STRING)));
+    org.apache.avro.Schema avroSchema = new org.apache.avro.Schema.Parser().parse(expectedSchema.toString());
+    GenericRecord record = new GenericRecordBuilder(avroSchema)
+      .set("id", 0)
+      .set("name", "alice")
+      .set("score", 1.0d)
+      .set("test", "value")
+      .build();
+
+    File avroFile = new File(TMP_FOLDER.newFolder(), "test.avro");
+    DatumWriter<GenericRecord> datumWriter = new GenericDatumWriter<>(avroSchema);
+    DataFileWriter<GenericRecord> dataFileWriter = new DataFileWriter<>(datumWriter);
+    dataFileWriter.create(avroSchema, avroFile);
+    dataFileWriter.append(record);
+    dataFileWriter.close();
+
+    // call detect schema
+    AvroInputFormatProvider.Conf conf = new AvroInputFormatProvider.Conf("test");
+    AvroInputFormatProvider formatProvider = new AvroInputFormatProvider(conf);
+    FormatContext formatContext = new FormatContext(new MockFailureCollector(), null);
+    SchemaDetector schemaDetector = new SchemaDetector(formatProvider);
+    schemaDetector.detectSchema(avroFile.getPath(), formatContext, Collections.emptyMap());
+    ValidationFailure validationFailure = formatContext.getFailureCollector().getValidationFailures().get(0);
+    Assert.assertEquals("Pathfield already exists in the schema", validationFailure.getMessage());
```

Comment:
generally speaking, we don't want to test the exact error message string. It makes the test brittle and it's not part of the contract. What we do want to check is that the causes for this validationFailure are as expected, with the path field as highlighted as the problematic config property.

Reply:
Checking config property pathField